### PR TITLE
hw5-3_done

### PR DESCRIPTION
**Pull request by vladkuzya**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
+                }
+            }
+            String[] result = resultString.trim().split(" ");
+            Arrays.sort(result);
+
+            return result[0] != "" ? result : new String[0];
+        } catch (FileNotFoundException e) {
+            System.out.println("File with name " + fileName + " not found!");
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/FileWork.java:48]: Don’t ignore exceptions. Re-throw them as RuntimeException(e.g. throw new RuntimeException(e)) if you don’t expect your code to fail here, or make necessary actions at the catch block to fix the problem.
https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_05.html

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -1,14 +1,52 @@
...
+                } else {
+                    return new String[0];
+                }
+            }
+            String[] result = resultString.trim().split(" ");
+            Arrays.sort(result);
+
+            return result[0] != "" ? result : new String[0];
+        } catch (FileNotFoundException e) {
+            System.out.println("File with name " + fileName + " not found!");
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/FileWork.java:46]: same here

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -1,14 +1,53 @@
...
-        return null;
+    public static final char LETTER = 'w';
+
+    public static String[] readFromFile(String fileName) {
+        try (FileReader fileReader = new FileReader(fileName)) {
+            Scanner scanner = new Scanner(fileReader);
+
+            String text = "";
+            while (scanner.hasNextLine()) {
+                text += scanner.nextLine() + " ";
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:26]: let's use StringBuilder and it's method append().
Tell me why that would be better in comment below?

File: src/main/java/core/basesyntax/FileWork.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/FileWork.java:46]: now we can delete that System.out.println

File: src/main/java/core/basesyntax/FileWork.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/FileWork.java:49]: same here about e.printStackTrace();

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -1,14 +1,53 @@
...
+            }
+            String[] words = text.split(" ");
+            String resultString = "";
+
+            int index = 0;
+            for (int i = 0; i < words.length; i++) {
+                if (words.length > 1) {
+                    if (words[i].toLowerCase().charAt(0) == LETTER) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:34]: we have string method to check if it's starts with character

File: src/main/java/core/basesyntax/FileWork.java (hunk 3 above)
Comment (vladkuzya) [reply src/main/java/core/basesyntax/FileWork.java:26]: StringBuilder работает намного быстрее String. И если бы у нас был текстовый файл на миллион строк и их нужно было все считать, то это была бы большая разница во времени исполнения между StringBuilder и String

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -1,14 +1,51 @@
...
+                text.append(scanner.nextLine() + " ");
+            }
+            String[] words = text.toString().split(" ");
+            String resultString = "";
+
+            int index = 0;
+            for (int i = 0; i < words.length; i++) {
+                if (words.length > 1) {
+                    if (words[i].toLowerCase().startsWith(LETTER)) {
+                        resultString += words[i].toLowerCase().replaceAll("[.), \\-'!?]", "") + " ";
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:35]: try to find regexp for that

File: src/main/java/core/basesyntax/FileWork.java [hunk 6]
```diff
@@ -1,14 +1,51 @@
...
+                }
+            }
+            String[] result = resultString.trim().split(" ");
+            Arrays.sort(result);
+
+            return result[0] != "" ? result : new String[0];
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:48]: If you are handling these two type of exceptions in the same way. You can just leave catch with IOException

File: src/main/java/core/basesyntax/FileWork.java [hunk 7]
```diff
@@ -1,14 +1,51 @@
...
+                        resultString += words[i].toLowerCase().replaceAll("[.), \\-'!?]", "") + " ";
+                    }
+                } else {
+                    return new String[0];
+                }
+            }
+            String[] result = resultString.trim().split(" ");
+            Arrays.sort(result);
+
+            return result[0] != "" ? result : new String[0];
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:44]: remove this check either here or on 33 line

**Pull request closed without merging**